pkg/postgres/walker/walker.go

```diff
@@ -188,7 +262,7 @@ func handleStruct(ctx context, schema *Schema, original reflect.Type) {
 		if strings.HasPrefix(structField.Name, "XXX") {
 			continue
 		}
-		opts := getPostgresOptions(structField.Tag.Get("sql"), len(schema.ReferencedSchema) == 0, ctx.ignorePK, ctx.ignoreUnique)
+		opts := getPostgresOptions(structField.Tag.Get("sql"), schema.Parent == nil, ctx.ignorePK, ctx.ignoreUnique, ctx.ignoreFKs)
```

Comment:
Not sure I follow?

Reply:
It is always guaranteed that non-nil Parents also means non-empty. I was thinking of making `topLevel` more correct. Nonetheless, the converse would mean a code generation bug which we can address when encountered.

Reply:
Hmm, I think there is no correctness issue here, because `topLevel` _only_ refers to the parent<->child relationship, which, after this PR, only refers to embedded messages. References via the `fk` tag are handled completely separately now.